### ISSUE TITLE
Disabling IPV6 in .NET Core

### ISSUE THREAD

**Issue by ansulhere** [dotnet/aspnetcore#57914]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

On .NET 6 alpine images, the setting to Disable IPv6 has no impact if the remote resource/URL does not have a Quad A DNS record.
Such an outbound request fails with System.Net.Http.HttpRequestException: Resource temporarily unavailable (remote_url)
Inner Exception : System.Net.Sockets.SocketException (11) : Resource temporarily unavailable.

The following setting **"System.Net.DisableIPv6" : true** has been tried and this does not seem to have any impact.

For Java applications calling the same remote url, the following setting disables ipv6 attempts and only ipv4 is attempted:
ENTRYPOINT ["java", "-Djava.net.preferIPv4Stack=true", "-Djava.security.egd=file:/dev/./urandom", "-jar", "/app.jar"]

**Comment by martincostello** [comment]
Looks like this issue should be transferred to [dotnet/runtime](https://github.com/dotnet/runtime) as it doesn't appear to be specific to ASP.NET Core.

**Comment by amcasey** [comment]
Yeah, let's start with runtime, though this may end up on docker-tools.

**Comment by avarakin** [comment]
I think it needs to be addressed in dotnet/runtime, as @martincostello indicated. The runtime itself should be capable of disabling IPV6 so we do not depend on container or OS configurations, which may not always work, as happened here.
P.S. I work for the company (WU) , where this issue was encountered and reported to @ansulhere as a support ticket.
